src/go/goblins/docker/docker.go

```diff
@@ -0,0 +1,209 @@
+		args := []string{"-taufs", opts, "none", mountDir}
+		commands = append(commands, exec.Command("mount", args...))
+	}
+
+	// Now mount volumes.
+	switch dockerVersion {
+	case 1:
+		return append(commands, mountVolumesDocker1(mountDir, rootDir, volumes)...), nil
+	case 2:
+		return append(commands, mountVolumesDocker2(mountDir, mountPoints)...), nil
+	default:
+		return nil, fmt.Errorf("Unsupported docker version: %d", dockerVersion)
+	}
+}
+
+func mountVolumesDocker1(mountDir string, rootDir string, volumes map[string]string) []*exec.Cmd {
+	commands := make([]*exec.Cmd, 0)
+	for mountpoint, storage := range volumes {
+		storageIhp := strings.TrimLeft(mountpoint, string(os.PathSeparator))
+		mountpointIhp := strings.TrimLeft(storage, string(os.PathSeparator))
+		storagePath := filepath.Join(rootDir, storageIhp)
+		volumeMountpoint := filepath.Join(mountDir, mountpointIhp)
+		args := []string{"--bind", "-oro", storagePath, volumeMountpoint}
+		commands = append(commands, exec.Command("mount", args...))
+	}
+	return commands
+}
+
+func mountVolumesDocker2(mountDir string, mountPoints []storageInfo) []*exec.Cmd {
+	commands := make([]*exec.Cmd, 0)
+	for _, s := range mountPoints {
+		srcMount := strings.TrimLeft(s.source, string(os.PathSeparator))
+		dstMount := strings.TrimLeft(s.destination, string(os.PathSeparator))
+		if srcMount == "" {
+			volumeName := s.name
+			srcMount = filepath.Join("docker", "volumes", volumeName, "_data")
+		}
+		storagePath := filepath.Join(mountDir, dstMount)
+		volumeMountpoint := filepath.Join(mountDir, dstMount)
+		args := []string{"--bind", "-oro", storagePath, volumeMountpoint}
+		commands = append(commands, exec.Command("mount", args...))
+	}
+	return commands
+}
+
+func overlayFsMountCommands(mountDir string, mountID string, dockerDir string,
+	storageMethod string, dockerVersion int) ([]*exec.Cmd, error) {
+	mountIDPath := filepath.Join(dockerDir, storageMethod, mountID)
+	const lowerdirName = "lower"
+	const upperdirName = "diff"
+
+	lowerDirPath := filepath.Join(mountIDPath, lowerdirName)
+	lowerContent, err := ioutil.ReadFile(lowerDirPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read lower content pointer: %s", err)
+	}
+
+	// There are differences between overlay and overlay2 here for the lower directory
+	if storageMethod == "overlay" {
+		lowerDirPath = filepath.Join(dockerDir, storageMethod, string(lowerContent), "root")
+	} else {
+		// For the overlay2 driver, the pointer to the 'lower directory' can be
+		// made of multiple layers.
+		// For that argument to be passed to the mount.overlay command, we need to
+		// reconstruct full paths to all these layers.
+		// ie: from 'abcd:0123' to '/var/lib/docker/abcd:/var/lib/docker/0123'
+		var paths []string
+		for _, lc := range strings.Split(string(lowerContent), ":") {
+			paths = append(paths, filepath.Join(dockerDir, storageMethod, lc))
+		}
+		lowerDirPath = strings.Join(paths, ":")
```

Comment:
Why not [`os.PathListSeparator`](https://golang.org/pkg/os/#pkg-constants) here?

Reply:
Sure.